File: lib/models/harcama_grubu_data.dart
class HarcamaGrubuData {
  final String groupName;
  final double changeRate;

  HarcamaGrubuData({
    required this.groupName,
    required this.changeRate,
  });

  factory HarcamaGrubuData.fromCsv(List<dynamic> row, int columnIndex) {
    return HarcamaGrubuData(
      groupName: row[1].toString().trim(),
      changeRate: double.tryParse(row[columnIndex].toString()) ?? 0.0,
    );
  }
}

class AnaGrupData {
  final String name;

  AnaGrupData({required this.name});
}

class HarcamaGrubuEndeksData {
  final String tarih;
  final double endeks;

  HarcamaGrubuEndeksData({
    required this.tarih,
    required this.endeks,
  });
}

class HarcamaGrubuAylikData {
  final String tarih;
  final double degisimOrani;

  HarcamaGrubuAylikData({
    required this.tarih,
    required this.degisimOrani,
  });
}

class HarcamaGrubuIstatistik {
  final double yillikDegisim;
  final double aylikDegisim;
  final String selectedGrup;

  HarcamaGrubuIstatistik({
    required this.yillikDegisim,
    required this.aylikDegisim,
    required this.selectedGrup,
  });
}

File: src/models/HarcamaGrubuData.ts
export interface HarcamaGrubuData {
  groupName: string;
  changeRate: number;
}

export interface AnaGrupData {
  name: string;
}

export interface HarcamaGrubuEndeksData {
  tarih: string;
  endeks: number;
}

export interface HarcamaGrubuAylikData {
  tarih: string;
  degisimOrani: number;
}

export interface HarcamaGrubuIstatistik {
  yillikDegisim: number;
  aylikDegisim: number;
  selectedGrup: string;
}

export class HarcamaGrubuDataModel implements HarcamaGrubuData {
  groupName: string;
  changeRate: number;

  constructor(groupName: string, changeRate: number) {
    this.groupName = groupName;
    this.changeRate = changeRate;
  }

  static fromCsv(row: any[], columnIndex: number): HarcamaGrubuData {
    return new HarcamaGrubuDataModel(
      row[1]?.toString().trim() || '',
      parseFloat(row[columnIndex]?.toString() || '0') || 0.0
    );
  }
}

